src/System.Net.NameResolution/src/System/Net/DNS.cs

```diff
@@ -157,6 +157,7 @@ internal static IPHostEntry InternalGetHostByAddress(IPAddress address, bool inc
         public static string GetHostName()
         {
             GlobalLog.Print("Dns.GetHostName");
+            NameResolutionPal.EnsureSocketsAreInitialized();
```

Comment:
@pgavlin Verified and this is not entirely possible as the PAL code was written.
For example, in any of the Async/APM APIs, the calls need to be made before Socket.OSSupportsIPv6.